packages/edit-site/src/components/global-styles/use-global-styles-output.js

```diff
@@ -211,6 +211,16 @@ function getStylesDeclarations(
 
 			const styleValue = get( blockStyles, pathToValue );
 
+			// Root-level padding styles don't currently support strings with CSS shorthand values.
+			// This may change: https://github.com/WordPress/gutenberg/issues/40132.
+			if (
+				( key === '--wp--style--root--padding' &&
+					typeof styleValue === 'string' ) ||
+				! useRootPaddingAlign
```

Comment:
For this condition, is it possible that more declarations will be skipped than desired if `! useRootPaddingAlign` evaluates to `true`? I was wondering if the condition is meant to be just for the `--wp--style--root--padding` key, like:

```
			if (
				key === '--wp--style--root--padding' &&
				( typeof styleValue === 'string' || ! useRootPaddingAlign )
			) {
				return declarations;
			}
```

Apologies if I'm not understanding the logic here correctly!

Reply:
It looks like `fontFamily` is one of the few ones that isn't using the style engine, so I found it a little challenging to work out an error state for this one, but I _think_ the following test seems to catch it:

```js
		it( 'Should output padding and font family properties if useRootPaddingAwareAlignments is disabled', () => {
			const blockStylesWithFontFamily = {
				...blockStyles,
				typography: {
					fontFamily: 'sans-serif',
				},
			};
			expect(
				getStylesDeclarations(
					blockStylesWithFontFamily,
					'body',
					false
				)
			).toEqual( [
				'font-family: sans-serif',
				'background-color: var(--wp--preset--color--light-green-cyan)',
				'padding-top: 33px',
				'padding-right: 33px',
				'padding-bottom: 33px',
				'padding-left: 33px',
			] );
		} );
```

Reply:
Oh, I've found a better real world example with the Duotone feature. In TwentyTwentyTwo, if you add the following to the `theme.json` file (from https://github.com/WordPress/gutenberg/pull/37727) to switch on a particular Duotone for image blocks:

```json
{
	"styles": {
		"blocks": {
			"core/image": {
				"filter": {
					"duotone": "var(--wp--preset--duotone--foreground-and-secondary)"
				}
			}
		}
	}
}
```

| Expected | Actual |
| --- | --- |
| <img width="1275" alt="image" src="https://user-images.githubusercontent.com/14988353/182771891-6c06f11c-e948-42d5-8933-87023a930d2a.png"> | <img width="1275" alt="image" src="https://user-images.githubusercontent.com/14988353/182772063-3ef0105b-3525-46ac-a66c-244dcf02fae7.png"> |

The above change appears to fix it.

Reply:
Oh, good catch! Yes, the parentheses should be around the OR as you suggested 😄